filesystem/fat32/testdata/mkfat32.sh

```diff
@@ -8,86 +8,92 @@ if [ -z "$INSTALLDIR" ]; then
   INSTALLDIR="$PWD"
 fi
 
+docker build -t go-diskfs-testdata-fat32 .
+
 mkdir -p $INSTALLDIR/dist
-cat <<"EOF" | docker run -i --rm -v $INSTALLDIR/dist:/data alpine:3.20
+cat <<"EOF" | docker run -i --rm -v $INSTALLDIR/dist:/data go-diskfs-testdata-fat32
 set -e
 set -x
 
-apk --update add dosfstools mtools sleuthkit
-dd if=/dev/zero of=/data/fat32.img bs=1M count=10
-mkfs.vfat -v -F 32 /data/fat32.img
-echo "mtools_skip_check=1" >> /etc/mtools.conf
-mmd -i /data/fat32.img ::/foo
-mmd -i /data/fat32.img ::/foo/bar
-echo 'Tenemos un archivo corto' > /CORTO1.TXT
-dd if=/dev/zero 'of=/Un archivo con nombre largo.dat' bs=1024 count=2
-dd if=/dev/zero 'of=/tercer_archivo' bs=1024 count=6
-dd if=/dev/zero 'of=/Un archivo con nombre largo.dat' bs=1024 count=7
-dd if=/dev/zero 'of=/some_long_embedded_nameא' bs=1024 count=7
-mcopy -i /data/fat32.img /CORTO1.TXT ::/
-mcopy -i /data/fat32.img '/Un archivo con nombre largo.dat' ::/
-mcopy -i /data/fat32.img '/tercer_archivo' ::/
-mcopy -i /data/fat32.img '/some_long_embedded_nameא' ::/foo/bar
-mmd -i /data/fat32.img ::/lower83
-mcopy -i /data/fat32.img /CORTO1.TXT ::/lower83/lower.low
-mcopy -i /data/fat32.img /CORTO1.TXT ::/lower83/lower.UPP
-mcopy -i /data/fat32.img /CORTO1.TXT ::/lower83/UPPER.low
+for n in $(echo "12 16 32"); do
+  out="/data/fat${n}"
+  mkdir ${out}
 
-i=0
-until [ $i -gt 75 ]; do mmd -i /data/fat32.img ::/foo/dir${i}; i=$(( $i+1 )); done
+  dd if=/dev/zero of=${out}/disk.img bs=1M count=$(( n + 2 ))
```

Comment:
This `$(( n + 2 ))` gives each FAT filesystem enough space to actually be a valid FAT partition.

In the original fixture the FAT32 image was given only `10mb`, which resulted in errors such as:

```
WARNING: Number of clusters for 32 bit FAT is less then suggested minimum.

Warning: Filesystem is FAT32 according to fat_length and fat32_length fields,
but has only 20132 clusters, less than the required minimum of 65525.
This may lead to problems on some systems.
```

Since it was a "technically incorrect" FAT32 filesystem.

Reply:
Yeah, but you always can override it.